hooks/pre/10-check_hostname.rb

```diff
@@ -0,0 +1,40 @@
+require 'facter'
+
+BASE = %q(If needed, change the hostname permanently via 'hostname' command and editing 
+appropriate configuration file.
+(e.g. on Red Hat systems /etc/sysconfig/network).
+If 'hostname -f' still returns unexpected result, check /etc/hosts and put
+hostname entry in the correct order, for example:
+ 
+  1.2.3.4 full.hostname.com full
+ 
+Fully qualified hostname must be the first entry on the line)
+
+DIFFERENT = %q(Output of 'facter fqdn' is different from 'hostname -f'
+ 
+Make sure above command gives the same output. )
+
+INVALID = %q(Output of 'hostname -f' does not seems to be valid FQDN
+Make sure above command gives fully qualified domain name. At least one
+dot must be present and underscores are not allowed. )
+
+UPCASE = %q(The hostname contains a a capital letter.
```

Comment:
I could see this check causing a bit of an unexpected headache.

the fusor_network plugin is writing the hostname like so:
    host { $fqdn:
      comment      => 'created by puppet class foreman::plugin::fusor_network',
      ip           => $ip,
      host_aliases => $hostname
    }

The problem is that if you set the hostname like so:
fusor-installer --hostname foo.Bar.redhat.com

You get an /etc/hosts record:
192.168.252.10	foo.Bar.redhat.com	foo	# created by puppet class foreman::plugin::fusor_network

Oops, ok, you screwed up, so:
fusor-installer --hostname foo.bar.redhat.com

but fqdn still resolves with the capital letter because of the hosts entry and so doesn't get fixed. You're stuck until you manually clean up now.

I think we should consider downcasing any input to prevent this issue.

We also do nothing to ensure that our record for this IP address is the first.

If someone has 192.168.252.10 my.localhost we'll just add another entry 192.168.252.10 whatever.we.entered.

We need to probably change fusor_network to purge the /etc/hosts and rewrite it with the values we specify if we're going to manipulate it at all.